src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java

```diff
@@ -22,19 +27,27 @@ public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
     private final BookSpecificationBuilder bookSpecificationBuilder;
+    private final CategoryRepository categoryRepository;
 
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
+        Category category = categoryRepository.findByName(requestDto.getCategoryName())
+                .orElseThrow(() -> new EntityNotFoundException("Category not found"));
+
         Book book = bookMapper.toModel(requestDto);
+        Set<Category> categories = new HashSet<>();
+        categories.add(category);
+        book.setCategories(categories);
         book = bookRepository.save(book);
         return bookMapper.toBookDto(book);
     }
 
     @Override
     public List<BookDto> findAll(Pageable pageable) {
-        return bookRepository.findAll(pageable).stream()
+        List<Book> books = bookRepository.findAll();
+        return books.stream()
                 .map(bookMapper::toBookDto)
-                .toList();
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                .toList();
```
Let’s  revert these changes